### PR TITLE
Feature/test params

### LINKED ISSUE #181: Tests still depend on referring to xmostest, when dependancy has been removed.
The functional tests in the tests sub-directory on DEV/XS3 still try to import xmostest. This is a) unnecessary as the code has been refactored, and b) fails, because the tests are now python3 and xmostest is python2.

### PR DESCRIPTION
**Pull request by xross**
~Initial work for a proposal to properly parameterise tests (starting with test_stall_basic)~

All tests now ported

The issue with previous implementation is that a single test would run different arch, bus-speed and other params, meaning if one of these fails all are marked as fail. In the proposed method they treated as separate tests by pysim. 

This also allows a full cross of the parameters to be used.

Going forward we would likely have a set of "smoke" and "extended/nightly" params as well as the "default" ones

I also marked some tests as known fails in PyTest and some formatting updates via Black.

Note, this don't yet cross or used the  params from PyTest - I'll work on that now (on a different PR)

Also fixes #181 

Note
 * test_sof_badcrc is incorrectly marked as pass (https://github.com/xmos/lib_xud/issues/182)
 * test_bulk_rx_traffic fails at full-speed (https://github.com/xmos/lib_xud/issues/183)

### CODE REVIEW

File: tests/Pyxsim/xmostest_subprocess.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ def platform_is_windows():
     use_shell = True
     # Windows version of Python 2.7 doesn't support SIGINT
     if sys.version_info < (3, 0):
-        raise TestError("Doesn't support Python version < 3, please upgrade to Python 3 or higher.")
+        raise TestError(
+            "Doesn't support Python version < 3, please upgrade to Python 3 or higher."
+        )
```
Comment (DaveAtkinsonXMOS) [inline tests/Pyxsim/xmostest_subprocess.py:90]: Isn't this true for all platforms, now? Maybe move this to somewhere generic?

**Review by DaveAtkinsonXMOS — COMMENTED** [review]
Looks good to me, one comment about the required python version...

**Review by kevinyeungxmos — COMMENTED** [review]
sounds good to me

File: tests/Pyxsim/xmostest_subprocess.py (hunk 1 above)
Comment (xross) [reply tests/Pyxsim/xmostest_subprocess.py:90]: I think this is something @kevinyeungxmos put in based on a comment from @mbanth (my changes were just auto-formatting via black)

Comment (xross) [reply tests/Pyxsim/xmostest_subprocess.py:90]: The whole pyxsim folder needs moving to another repo.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import pytest
```
Comment (mbanth) [inline tests/conftest.py:1]: Missing copyright and license comments

**Review by mbanth — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (xross) [reply tests/conftest.py:1]: thanks

**Comment by xross** [comment]
I've now ported the rest of the tests.

**Comment by xross** [comment]
@mbanth are you happy with this one now please?

**Review by mbanth — APPROVED** [review]

**Pull request merged**